crates/ruma-identifiers/src/mxc_uri.rs

```diff
@@ -30,6 +30,11 @@ impl MxcUri {
 
     /// If this is a valid MXC URI, returns a `(server_name, media_id)` tuple.
     pub fn parts(&self) -> Option<(&ServerName, &str)> {
+        self.parts_err().ok()
+    }
+
+    /// Similar to [MxcUri::parts], except returns the error if this is not a valid MXC URI.
+    pub fn parts_err(&self) -> Result<(&ServerName, &str), MxcUriError> {
```

Comment:
Since this PR is a breaking change anyways, let's just change the existing `.parts()` fn's return type rather than introducing a new one. Where needed, one can call `.ok()` on that.

Reply:
Will do